Лабораторная работа 4

listopadnyayap opened the pull request.

listopadnyayap committed:
Лабораторная работа 4

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -0,0 +1,23 @@
+    for symbol in str_: #Перебор всех символов в строке
+        if symbol.isalpha() and symbol not in symbol_dict: #Условие, при котором символ является алфавитным и ещё не включен в словарь
+            if str_.count(symbol)>=1:
+                symbol_dict[symbol]=str_.count(symbol) #Добавление новых элементов в словарь
```
Если вы считате количество вхождений конкретного символа в строку с помощью встроенного метода, для чего вам эта проверка - if str_.count(symbol)>=1?

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -0,0 +1,23 @@
+print(get_count_char(str_))
+
+
+
```
Вы не реализовали вторую часть задания "Напишите вторую функцию, которая принимает словарь символов. Функция должна вернуть новый словарь, где количество каждого
элемента заменено на процентное отношение ко всем символам."

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -0,0 +1,23 @@
+    В качестве разделителя для встроенного метода split будет выбран символ пробела. На выходе мы получим список отдельных слов. 
+    Далее нужно отсортировать слова в алфавитном порядке, а после сортировки склеить их с помощью метода строк join. Приступим!!!!
+"""
+str_=''.join(main_str.lower().split(' '))
```
Для чего вы выполняете эту операцию? Приведение к нижнему регистру должна выполнять сама функция

ksigorodetskaya commented on Лабораторная работа 4/task_3.py:
```diff
@@ -0,0 +1,13 @@
+        list_.pop(-1)
+        return list_
+
+
```
Если в функцию будет передан отрицательный индекс верно ли отработает ваша функция?

ksigorodetskaya replied:
Как отработает ваша функция вызов
print(delete([0, 1, 1, 2, 3], index=-1)) ?

listopadnyayap committed:
Update task_2.py

Добавлена вторая функция для подсчёта процентного отношения и исправлены ошибки согласно замечаниям

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -1,23 +1,19 @@
+symbol_dict=dict()
 def get_count_char(str_):
-    symbol_dict=dict()
+    str_ = ''.join(main_str.lower().split(' '))
```
Причем здесь main_str? Вы должны передавать в вашу функцию строку (вы определили аргумент у вашей функции) и она должна работать с тем, что вы ей передали, а не с тем что находится за ее пределами. И опять вопрос: зачем вы выполняете разбиение на слова?

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -1,23 +1,19 @@
+symbol_dict=dict()
 def get_count_char(str_):
-    symbol_dict=dict()
```
Из каких соображений вы перенесли symbol_dict=dict() за пределы функции?
Словарь формируется в функции и функция его возвращает

ksigorodetskaya commented on Лабораторная работа 4/task_2.py:
```diff
@@ -1,23 +1,19 @@
-
-
-
+print(get_count_char(main_str))
```
Как должен выглядеть вызов функции  get_percent?
